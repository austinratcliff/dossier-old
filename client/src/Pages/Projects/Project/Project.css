.Project {
  font-family: 'IBM Plex Mono', monospace;
  margin-top: 25px;
}

.Project:last-child {
  margin-bottom: 25px;
}

.Project .name {
  font-size: calc(10px + 2vmin);
  margin: 0;
}

.Project .highlight-hover:hover {
  background: linear-gradient(180deg, rgba(255, 255, 255, 0) 50%, #92A8D1 50%);
  cursor: pointer;
}

.modal-backdrop {
  background-color: #111 !important;
}

.modal-content {
  border: 3px solid #111 !important;
  border-radius: 0 !important;
  background-color: white !important;
  color: #111 !important;
}

.modal-header {
  border-bottom: 3px solid #111 !important;
  font-family: 'Ibm Plex Mono', monospace !important;
}

.modal-body {
  font-family: 'IBM Plex Sans', sans-serif !important;
}

.modal-body h1,
.modal-body h2,
.modal-body h3,
.modal-body h4,
.modal-body h5,
.modal-body h6 {
  font-family: 'Ibm Plex Mono', monospace !important;
}

.modal-body a {
  background: linear-gradient(180deg, rgba(255, 255, 255, 0) 50%, #92A8D1 50%) !important;
  color: #111 !important;
}

.modal-body a:hover {
  color: #111 !important;
  text-decoration: none !important;
}

.modal-body img {
  display: block !important;
  margin: 0 auto !important;
}

.modal-body pre {
  border: 1px solid #111 !important;
  background-color: #111 !important;
  color: white !important;
  font-weight: bold !important;
  letter-spacing: 1.5px !important;
  display: flex !important;
  justify-content: center !important;
  align-self: center !important;
  width: 100% !important;
  padding: 15px 0 !important;
}

.modal-body code {
  font-family: 'Ibm Plex Mono', monospace !important;
}

.modal-body::-webkit-scrollbar {
  display: none !important;
}

.modal-footer {
  border-top: 3px solid #111 !important;
  font-family: 'IBM Plex Sans', sans-serif !important;
  justify-content: flex-start !important;
}

.modal-footer .url {
  color: #111 !important;
  text-decoration: none !important;
  margin-right: 10px !important;
}

.modal-footer .badge {
  margin-right: 5px !important;
}

.modal-footer .CREATOR {
  background-color: #92A8D1 !important;
  color: white !important;
}

.modal-footer .CONTRIBUTOR {
  background-color: #F7CAC9 !important;
  color: white !important;
}

.modal-footer .ACTIVE {
  background-color: #5A5B9F !important;
  color: white !important;
}

.modal-footer .INACTIVE {
  background-color: #964F4C !important;
  color: white !important;
}
