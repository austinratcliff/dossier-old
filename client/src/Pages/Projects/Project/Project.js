import React, { useState } from 'react'
import Badge from 'react-bootstrap/Badge'
import Modal from 'react-bootstrap/Modal'
import ReactMarkdown from 'react-markdown'
import './Project.css'

const Project = props => {
  const [show, setShow] = useState({ [props.project._id]: false })

  const handleShow = () => setShow({ ...show, [props.project._id]: true })
  const handleHide = () => setShow({ ...show, [props.project._id]: false })

  return (
    <div className="Project">
      <h2
        className="name"
        onClick={handleShow}
      >
        <span className="highlight-hover">
          {props.project.name}
        </span>
      </h2>
      <Modal
        show={show[props.project._id]}
        onHide={handleHide}
        animation={false}
        size="lg"
        centered
        scrollable
      >
        <Modal.Header>
          <Modal.Title>
            {props.project.name}
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <ReactMarkdown source={props.project.description} />
        </Modal.Body>
        <Modal.Footer>
          <a
            className="url"
            href={props.project.projectURL}
            target="_blank"
            rel="noopener noreferrer"
          >
            <i
              className="fas fa-laptop-code"
              title="project"
            /> Visit
          </a>
          <a
            className="url"
            href={props.project.githubURL}
            target="_blank"
            rel="noopener noreferrer"
          >
            <i
              className="fab fa-github"
              title="github"
            /> GitHub
          </a>
          {
            props.project.badges.map(badge => <Badge key={badge} className={badge} pill>{badge}</Badge>)
          }
        </Modal.Footer>
      </Modal>
    </div>
  )
}

export default Project
