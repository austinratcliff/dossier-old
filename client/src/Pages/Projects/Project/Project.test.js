import React from 'react'
import ReactDOM from 'react-dom'
import ReactTestRenderer from 'react-test-renderer'
import Project from './Project'

describe('Project component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Project />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('matches snapshot', () => {
    const component = ReactTestRenderer.create(<Project />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
