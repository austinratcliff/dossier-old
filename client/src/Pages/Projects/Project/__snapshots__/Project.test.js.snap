// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component matches snapshot 1`] = `
<div
  className="Project"
/>
`;
